Tests/TSCUtilityTests/VersionTests.swift

```diff
@@ -44,10 +277,60 @@ class VersionTests: XCTestCase {
 
         XCTAssertEqual(Set([Version(1,2,3)]), Set([Version(1,2,3)]))
         XCTAssertNotEqual(Set([Version(1,2,3)]), Set([Version(1,2,3, prereleaseIdentifiers: ["alpha"])]))
-        XCTAssertNotEqual(Set([Version(1,2,3)]), Set([Version(1,2,3, buildMetadataIdentifiers: ["1011"])]))
+        XCTAssertEqual(Set([Version(1,2,3)]), Set([Version(1,2,3, buildMetadataIdentifiers: ["1011"])]))
```

Comment:
Now that two version that differ only by their build metadata identifiers are considered equal, they can not exist in the same set, or as keys in the same dictionary, etc. Would this be considered a correct behaviour?

Reply:
I'm not sure whether it would or not — could this cause problems for existing packages?  I suppose in that case there would already have been an error about there being two versions of a single package in the same graph, when according to semver rules those should always have been treated the same.  So if I understand correctly this would be safe because it's more permissive than it would have been in the past?

Reply:
I'm not sure whether that really answers anything — I'm trying to understand whether there could be anything here that causes existing packages to break.